src/functionals.jl

```diff
@@ -1,34 +1,39 @@
-function getEndpoints(distr::UnivariateDistribution, epsilon::Real)
-    (left,right) = map(x -> quantile(distr,x), (0,1))
-    leftEnd = left!=-Inf ? left : quantile(distr, epsilon)
-    rightEnd = right!=-Inf ? right : quantile(distr, 1-epsilon)
-    (leftEnd, rightEnd)
-end
-
-function expectation(distr::ContinuousUnivariateDistribution, g::Function, epsilon::Real)
-    f = x->pdf(distr,x)
-    (leftEnd, rightEnd) = getEndpoints(distr, epsilon)
-    quadgk(x -> f(x)*g(x), leftEnd, rightEnd)[1]
+function expectation(distr::ContinuousUnivariateDistribution, g::Function; kwargs...)
+    return first(quadgk(x -> pdf(distr, x) * g(x), extrema(distr)...; kwargs...))
 end
 
 ## Assuming that discrete distributions only take integer values.
 function expectation(distr::DiscreteUnivariateDistribution, g::Function, epsilon::Real)
-    f = x->pdf(distr,x)
-    (leftEnd, rightEnd) = getEndpoints(distr, epsilon)
-    sum(x -> f(x)*g(x), leftEnd:rightEnd)
+    mindist, maxdist = extrema(distr)
+    # We want to avoid taking values up to infinity
+    minval = isfinite(mindist) ? mindist : quantile(distr, epsilon)
+    maxval = isfinite(maxdist) ? maxdist : quantile(distr, 1 - epsilon)
+    return sum(x -> pdf(distr, x) * g(x), minval:maxval)
+end
+
+function expectation(distr::DiscreteUnivariateDistribution, g::Function; epsilon::Real=1e-10, kwargs...)
+    return expectation(distr, g, ϵ)
 end
```

Comment:
Maybe just change the `expectation` definition above to
```julia
function expectation(distr::DiscreteUnivariateDistribution, g::Function; epsilon::Real=1e-10)
...
```
? In any case, the unicode epsilon is not defined.